### PR TITLE
Added MP3Header#duration

### PR DESCRIPTION
**Pull request by njh**
A small change to add a MP3Header#duration method

### CODE REVIEW

**Comment by njh** [comment]
Background is that I am working on a ruby script to convert an MP3 file into an HLS stream:
https://github.com/njh/mp3-to-hls

Being able to easily get the duration of a frame would make much code simpler.

**Comment by njh** [comment]
ps. thank you for writing this gem!

File: spec/mp3file/mp3_header_spec.rb [hunk 1]
```diff
@@ -333,4 +333,21 @@
...
+      [ "MPEG 1 Layer II 48 kHz",    0xFD, 0x34, 0.024 ],
+      [ "MPEG 1 Layer III 44.1 kHz", 0xFB, 0x53, 0.026122448979591838 ],
+      [ "MPEG 2 Layer II 16 kHz",    0xF5, 0x48, 0.072 ],
+      [ "MPEG 2 Layer III 24 kHz",   0xF3, 0xD5, 0.024 ],
+    ]
+    combinations.each do |name, b2, b3, duration|
+      it "for #{name}, returns #{duration} seconds" do
+        io = create_io([ 0xFF, b2, b3, 0xC1 ])
+        h = Mp3file::MP3Header.new(io)
+        expect(h.duration).to eq(duration)
```
Comment (ahwatts) [inline spec/mp3file/mp3_header_spec.rb]: This is a floating-point comparison, so the `be_within` matcher might be more appropriate here.

Given that the sample counts are on the order of 100 samples / frame, and the sample rates are on the order of 10 kHz, we would expect the durations to be on the order of (100 / 10000) = 1e-2 sec.

Could you change the expected durations to be e.g. (for a 44.1kHz MPEG-1 Layer III) `1152.0 / 44100.0` to make it more obvious where they come from, and then expect the code's answer to `be_within(1e-5).of(duration)`, ensuring that we're within 0.1% of the correct duration.

**Review by ahwatts — CHANGES_REQUESTED** [review]
Thanks for the PR!

In general, it looks good to me. Just a little clarity change to the test, and I'll merge it.

**Review by ahwatts — APPROVED** [review]

**Comment by ahwatts** [comment]
I cut a new release with these changes: v1.3.0

**Comment by njh** [comment]
That is awesome, thank you very much!

**Pull request merged**